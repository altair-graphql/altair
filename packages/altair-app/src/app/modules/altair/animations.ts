import { trigger, transition, style, animate } from '@angular/animations';

export const fadeInOutAnimationTrigger = trigger('fadeInOutAnimation', [
  transition(':enter', [
    style({ opacity: 0 }),
    animate('.3s ease', style({ opacity: 1 })),
  ]),
  transition(':leave', [
    style({ opacity: 1 }),
    animate('.3s ease', style({ opacity: 0 })),
  ]),
]);
export const fadeGridGrowInOutAnimationTrigger = trigger(
  'fadeGridGrowInOutAnimation',
  [
    transition(':enter', [
      style({ opacity: 0, gridTemplateRows: '0fr' }),
      animate('.3s ease', style({ opacity: 1, gridTemplateRows: '1fr' })),
    ]),
    transition(':leave', [
      style({ opacity: 1, gridTemplateRows: '1fr'}),
      animate('.3s ease', style({ opacity: 0, gridTemplateRows: '0fr'})),
    ]),
  ]
);
