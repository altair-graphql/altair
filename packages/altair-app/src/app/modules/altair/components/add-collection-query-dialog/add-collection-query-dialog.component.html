<app-dialog
  [showDialog]="showDialog"
  [heading]="'SAVE_TO_COLLECTION_HEADER' | translate"
  [subheading]="'SAVE_TO_COLLCTION_SUBTEXT' | translate"
  (toggleDialog)="toggleDialogChange.emit($event)"
  (saveChange)="onSaveChange()"
>
  <div class="add-to-collection-wrapper">
    <div class="app-dialog-section">
      {{ 'SAVE_TO_COLLECTION_NAME_OF_QUERY' | translate }}
      <input
        type="text"
        class="dialog-input"
        [(ngModel)]="newCollectionQueryTitle"
        attr.aria-label="{{ 'SAVE_TO_COLLECTION_NAME_OF_QUERY' | translate }}"
        data-test-id="collection-query-name"
      />
    </div>
    <div class="app-dialog-section">
      <nz-form-label>{{
        'SAVE_TO_COLLECTION_SELECT_COLLECTION' | translate
      }}</nz-form-label>
      <nz-form-control>
        <nz-select
          class="dialog-block"
          [(ngModel)]="collectionId"
          name="collection"
          nzPlaceHolder="Choose"
        >
          <nz-option
            *ngFor="let collection of collections; trackBy: trackById"
            [nzValue]="collection.id"
            [nzLabel]="collection.title"
          ></nz-option>
          <nz-option
            nzValue="-1"
            [nzLabel]="
              'SAVE_TO_COLLECTION_OR_CREATE_NEW_COLLECTION_LABEL' | translate
            "
          ></nz-option>
        </nz-select>
      </nz-form-control>
    </div>
    <ng-container *ngIf="isNewCollection()">
      <div class="app-dialog-section">
        {{ 'SAVE_TO_COLLECTION_COLLECTION_NAME' | translate }}
        <input
          type="text"
          class="dialog-input"
          [(ngModel)]="newCollectionTitle"
          placeholder="Collection name"
          attr.aria-label="{{
            'SAVE_TO_COLLECTION_COLLECTION_NAME' | translate
          }}"
          data-test-id="new-collection-name"
        />
      </div>
      <div class="app-dialog-section">
        <nz-form-label> Save collection to </nz-form-label>
        <nz-form-control>
          <nz-tree-select
            [nzNodes]="collectionNodes"
            [nzNotFoundContent]="'No collections found'"
            nzShowSearch
            nzPlaceHolder="Select collection"
            [(ngModel)]="newCollectionParentCollectionId"
            (ngModelChange)="newCollectionParentCollectionIdChange.emit($event)"
          ></nz-tree-select>
        </nz-form-control>
      </div>
      <!-- if logged in, and not saving to a parent collection, then show workspaces option -->
      <div
        class="app-dialog-section"
        *ngIf="
          loggedIn && newCollectionParentCollectionId === parentCollectionRootId
        "
      >
        <nz-form-label>{{
          'SAVE_TO_COLLECTION_SELECT_WORKSPACE' | translate
        }}</nz-form-label>
        <nz-form-control>
          <nz-select
            class="dialog-block"
            [(ngModel)]="workspaceId"
            name="workspace"
            nzPlaceHolder="Choose"
          >
            <nz-option
              *ngFor="let workspace of workspacesOptions(); trackBy: trackById"
              [nzValue]="workspace.id"
              [nzLabel]="workspace.label"
            ></nz-option>
          </nz-select>
        </nz-form-control>
      </div>
    </ng-container>
  </div>
</app-dialog>
