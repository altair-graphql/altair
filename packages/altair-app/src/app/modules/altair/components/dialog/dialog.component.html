<nz-modal
  [nzVisible]="showDialog"
  (nzVisibleChange)="toggleDialog.next($event)"
  [nzCloseIcon]="'close-circle'"
  [nzTitle]="showHeader ? modalTitle : null"
  [nzContent]="modalContent"
  [nzFooter]="showFooter ? modalFooter : null"
  [nzWidth]="width"
  (nzOnCancel)="toggleDialog.next()"
>
  <ng-template #modalTitle>
    <div class="app-dialog-header">
      <div class="app-dialog-title">{{ heading }}</div>
      {{ subheading }}
    </div>
  </ng-template>

  <ng-template #modalContent>
    <ng-content></ng-content>
  </ng-template>

  <ng-template #modalFooter>
    @if (showFooter) {
      <div class="app-dialog-footer">
        <button class="btn btn--primary right" (click)="onClickSave($event)">
          {{ 'SAVE_BUTTON' | translate }}
        </button>
      </div>
    }
  </ng-template>
</nz-modal>
