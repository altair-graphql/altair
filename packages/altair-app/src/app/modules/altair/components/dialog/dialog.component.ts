import { Component, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-dialog',
  templateUrl: './dialog.component.html',
  styleUrls: ['./dialog.component.scss'],
})
export class DialogComponent {
  @Input() showDialog = false;
  @Input() heading = '[DIALOG_HEADING]';
  @Input() subheading = '[DIALOG_SUBHEADING]';
  @Input() showFooter = true;
  @Output() toggleDialog = new EventEmitter();
  @Output() saveChange = new EventEmitter();
  constructor() {}

  onClickSave(e: MouseEvent) {
    this.toggleDialog.emit(e);
    this.saveChange.emit(e);
  }
}
