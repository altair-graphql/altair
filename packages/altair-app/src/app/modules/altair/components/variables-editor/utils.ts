import {
  getNullableType,
  GraphQLInputObjectType,
  GraphQLInputType,
  isEnumType,
  isLeafType,
  isListType,
  isNullableType,
  isSpecifiedScalarType,
} from 'graphql';
import {
  JSONSchema7,
  JSONSchema7Definition,
  JSONSchema7Type,
  JSONSchema7TypeName,
} from 'json-schema';

/**
 * Maximum depth to traverse the GraphQL type tree for variable type resolution.
 * This is to prevent slow performance because of resolving the variable types and/or infinite recursion.
 */
const MAX_DEPTH = 4;

export type VariableToType = Record<string, GraphQLInputType>;

const specifiedScalarTypeToJSONSchemaType: Record<string, JSONSchema7TypeName> =
  {
    String: 'string',
    Int: 'integer',
    Float: 'number',
    Boolean: 'boolean',
    ID: 'string',
  };
const wrapWithAnyOfNull = (def: JSONSchema7, wrap?: boolean) => {
  if (!wrap) {
    return def;
  }
  return {
    anyOf: [def, { type: 'null' as JSONSchema7TypeName }],
  };
};
export const graphqlInputTypeToJsonSchemaType = (
  originalType: GraphQLInputType,
  depth = 0,
  defaultValue?: JSONSchema7Type
): JSONSchema7 => {
  if (depth > MAX_DEPTH) {
    // Max depth reached. Default to string
    return { type: 'string' };
  }

  const type = getNullableType(originalType);

  // Handle enums
  if (isEnumType(type)) {
    return wrapWithAnyOfNull(
      {
        type: 'string',
        description: type.description ?? undefined,
        default: defaultValue,
        anyOf: type.getValues().map((v) => ({
          enum: [v.value],
          description: v.description ?? undefined,
        })),
      },
      isNullableType(originalType)
    );
  }
  if (isLeafType(type)) {
    if (isSpecifiedScalarType(type)) {
      return wrapWithAnyOfNull(
        {
          type:
            specifiedScalarTypeToJSONSchemaType[type.toString()] ?? 'string',
          description: type.description ?? undefined,
          default: defaultValue,
        },
        isNullableType(originalType)
      );
    }
    // custom scalar types default to string
    return wrapWithAnyOfNull(
      {
        type: 'string',
        description: type.description ?? undefined,
      },
      isNullableType(originalType)
    );
  }
  // Handle lists
  if (isListType(type)) {
    return wrapWithAnyOfNull(
      {
        type: 'array',
        items: graphqlInputTypeToJsonSchemaType(type.ofType, depth + 1),
        description: type.ofType?.description ?? undefined,
        default: defaultValue,
      },
      isNullableType(originalType)
    );
  }

  if (type instanceof GraphQLInputObjectType) {
    return {
      type: 'object',
      description: type.description ?? undefined,
      properties: Object.entries(type.getFields()).reduce(
        (acc, [key, field]) => {
          acc[key] = graphqlInputTypeToJsonSchemaType(
            field.type,
            depth + 1,
            field.defaultValue
          );
          return acc;
        },
        {} as Record<string, JSONSchema7Definition>
      ),
    };
  }

  // Unknown case. Default to string
  return { type: 'string' };
};

/*
 * Converts a VariableToType object to a JSON Schema object.
 * This is used to generate a JSON Schema for the variables editor.
 */
export const vttToJsonSchema = (vtt?: VariableToType): JSONSchema7 => {
  if (!vtt) {
    return {
      type: 'object',
      properties: {},
    };
  }
  return {
    type: 'object',
    properties: Object.entries(vtt).reduce((acc, [key, type]) => {
      acc[key] = graphqlInputTypeToJsonSchemaType(type);
      return acc;
    }, {} as Record<string, JSONSchema7Definition>),
  };
};
