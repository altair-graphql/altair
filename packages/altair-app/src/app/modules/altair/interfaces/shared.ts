export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
