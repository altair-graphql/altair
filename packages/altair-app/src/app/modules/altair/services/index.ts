export { ApiService } from './api/api.service';
export { GqlService } from './gql/gql.service';
export { DbService } from './db.service';
export { WindowService } from './window.service';
export { NotifyService } from './notify/notify.service';
export { DonationService } from './donation.service';
export { ElectronAppService } from './electron-app/electron-app.service';
export { KeybinderService } from './keybinder/keybinder.service';
export { StorageService } from './storage/storage.service';
export { QueryCollectionService } from './query-collection/query-collection.service';
export { EnvironmentService } from './environment/environment.service';
export { PluginRegistryService } from './plugin/plugin-registry.service';
export { PluginEventService } from './plugin/plugin-event.service';
export { PreRequestService } from './pre-request/pre-request.service';
export { ThemeRegistryService } from './theme/theme-registry.service';
export { SubscriptionProviderRegistryService } from './subscriptions/subscription-provider-registry.service';
export { PluginContextService } from './plugin/context/plugin-context.service';
export { QueryService } from './query/query.service';
export { AccountService } from './account/account.service';
