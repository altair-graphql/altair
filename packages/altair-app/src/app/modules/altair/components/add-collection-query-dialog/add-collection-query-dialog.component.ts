import { Component, Output, EventEmitter, Input, OnChanges, SimpleChanges } from '@angular/core';
import { IQueryCollection, IQueryCollectionTree } from 'altair-graphql-core/build/types/state/collection.interfaces';
import { NzTreeNodeOptions } from 'ng-zorro-antd/tree';
import { Observable } from 'rxjs';
import { QueryCollectionService } from '../../services';

@Component({
  selector: 'app-add-collection-query-dialog',
  templateUrl: './add-collection-query-dialog.component.html',
  styles: [
  ]
})
export class AddCollectionQueryDialogComponent implements OnChanges {

  @Input() showDialog = false;
  @Input() windowTitle = '';
  @Input() collections: IQueryCollection[] = [];

  @Output() toggleDialogChange = new EventEmitter();
  @Output() createCollectionAndSaveQueryToCollectionChange = new EventEmitter();
  @Output() saveQueryToCollectionChange = new EventEmitter();
  @Output() newCollectionParentCollectionIdChange = new EventEmitter();

  newCollectionQueryTitle = this.windowTitle;
  newCollectionTitle = '';
  collectionId = null;
  newCollectionParentCollectionId = 0; // 0 for root
  collectionNodes: NzTreeNodeOptions[];

  constructor(
    private collectionService: QueryCollectionService,
  ) {
  }
  ngOnChanges(changes: SimpleChanges) {
    if (changes && changes.windowTitle && changes.windowTitle.currentValue) {
      this.windowTitle = changes.windowTitle.currentValue;
      this.reset();
    }
    if (changes?.collections?.currentValue) {
      this.setTreeNodes(changes.collections.currentValue);
    }
  }

  createCollectionAndSaveQueryToCollection() {
    this.createCollectionAndSaveQueryToCollectionChange.emit({
      queryName: this.newCollectionQueryTitle,
      collectionName: this.newCollectionTitle,
      parentCollectionId: this.newCollectionParentCollectionId,
    });

    this.reset();
  }

  saveQueryToCollection() {
    this.saveQueryToCollectionChange.emit({
      queryName: this.newCollectionQueryTitle,
      collectionId: this.collectionId
    });

    this.reset();
  }

  onSaveChange() {
    if (this.collectionId && this.collectionId !== -1) {
      return this.saveQueryToCollection();
    }
    return this.createCollectionAndSaveQueryToCollection();
  }

  reset() {
    this.newCollectionQueryTitle = this.windowTitle;
    this.newCollectionTitle = '';
    this.collectionId = null;
  }

  trackById(index: number, item: { id: string }) {
    return item.id;
  }

  setTreeNodes(collections: IQueryCollection[]) {
    const collectionTree = this.collectionService.getCollectionTrees(collections);
    this.collectionNodes = collectionTree.map(tree => this.collectionTreeToNzTreeNode(tree));
  }

  collectionTreeToNzTreeNode(collectionTree: IQueryCollectionTree): NzTreeNodeOptions {
    return {
      title: collectionTree.title,
      key: `${collectionTree.id}`,
      children: collectionTree.collections?.map(ct => this.collectionTreeToNzTreeNode(ct)),
    };
  }
}
