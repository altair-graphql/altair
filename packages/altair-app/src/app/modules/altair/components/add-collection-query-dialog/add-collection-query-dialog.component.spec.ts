import { expect } from '@jest/globals';

import { AddCollectionQueryDialogComponent } from './add-collection-query-dialog.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule } from '@angular/forms';
import { SharedModule } from '../../modules/shared/shared.module';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { mount } from '../../../../../testing/utils';
import { NgxTestWrapper } from 'testing/wrapper';
import { DialogComponent } from '../dialog/dialog.component';
import { QueryCollectionService, StorageService } from '../../services';
import { mock } from '../../../../../testing';

describe('AddCollectionQueryDialogComponent', () => {
  let wrapper: NgxTestWrapper<AddCollectionQueryDialogComponent>;

  beforeEach(async () => {
    wrapper = await mount({
      component: AddCollectionQueryDialogComponent,
      declarations: [AddCollectionQueryDialogComponent],
      providers: [QueryCollectionService, StorageService],
      imports: [BrowserAnimationsModule, FormsModule, SharedModule],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  /**
   * Inputs
   * - showDialog
   * - windowTitle
   * - collections
   * - event: toggleDialog
   * - event: createCollectionAndSaveQuery
   * - event: saveQueryToCollection
   */

  it('should render correctly', () => {
    expect(wrapper.component.nativeElement).toMatchSnapshot();
  });

  it('should set showDialog on app-dialog with [showDialog]', async () => {
    const appDialog = wrapper.find<DialogComponent>('app-dialog');
    wrapper.setProps({ showDialog: true });

    expect(appDialog.componentInstance.showDialog).toBe(true);
  });

  it('should render correctly with [windowTitle]', async () => {
    wrapper.setProps({ windowTitle: 'my title' });
    await wrapper.nextTick();

    expect(wrapper.component.nativeElement).toMatchSnapshot();
  });

  it('should render [windowTitle] in input', async () => {
    wrapper.setProps({ windowTitle: 'my title' });
    const input = wrapper.find('[data-test-id="collection-query-name"]');

    await wrapper.nextTick();

    expect(input.component.nativeElement.value).toBe('my title');
  });

  it('should render correctly with [collections]', async () => {
    wrapper.setProps({
      collections: [
        {
          title: 'query 1',
          id: 1,
          queries: [],
        },
        {
          title: 'query 2',
          id: 2,
          queries: [],
        },
      ],
    });

    await wrapper.nextTick();

    expect(wrapper.component.nativeElement).toMatchSnapshot();
  });

  it('should render new collection name input when selected collection is -1', async () => {
    wrapper.setProps({
      collections: [
        {
          title: 'query 1',
          id: 1,
          queries: [],
        },
        {
          title: 'query 2',
          id: 2,
          queries: [],
        },
      ],
    });

    const select = wrapper.find('nz-select');
    select.setValue(-1);

    await wrapper.nextTick();

    const newCollectionName = wrapper.find(
      '[data-test-id="new-collection-name"]'
    );

    expect(wrapper.componentInstance.collectionId).toBe(-1);
    expect(newCollectionName.exists()).toBeTruthy();
  });

  it('should emit createCollectionAndSaveQueryChange event when form is saved with new collection selected', async () => {
    wrapper.setProps({
      collections: [
        {
          title: 'query 1',
          id: 1,
          queries: [],
        },
        {
          title: 'query 2',
          id: 2,
          queries: [],
        },
      ],
    });

    const queryNameInput = wrapper.find(
      '[data-test-id="collection-query-name"]'
    );
    queryNameInput.setValue('my query name');

    const select = wrapper.find('nz-select');
    select.setValue(-1);

    await wrapper.nextTick();

    const newCollectionNameInput = wrapper.find(
      '[data-test-id="new-collection-name"]'
    );
    newCollectionNameInput.setValue('my new collection name');

    const appDialog = wrapper.find('app-dialog');
    appDialog.emit('saveChange');

    expect(
      wrapper.emitted('createCollectionAndSaveQueryToCollectionChange')
    ).toBeTruthy();
    expect(
      wrapper.emitted('createCollectionAndSaveQueryToCollectionChange')![0][0]
    ).toEqual({
      queryName: 'my query name',
      collectionName: 'my new collection name',
      parentCollectionId: 0,
    });
  });

  it('should emit [saveQueryToCollectionChange] event when form is saved with existing collection', async () => {
    wrapper.setProps({
      collections: [
        {
          title: 'query 1',
          id: 1,
          queries: [],
        },
        {
          title: 'query 2',
          id: 2,
          queries: [],
        },
      ],
    });

    const queryNameInput = wrapper.find(
      '[data-test-id="collection-query-name"]'
    );
    queryNameInput.setValue('my query name');

    const select = wrapper.find('nz-select');
    select.setValue(2);

    const appDialog = wrapper.find('app-dialog');
    appDialog.emit('saveChange');

    expect(wrapper.emitted('saveQueryToCollectionChange')).toBeTruthy();
    expect(wrapper.emitted('saveQueryToCollectionChange')![0][0]).toEqual({
      queryName: 'my query name',
      collectionId: 2,
    });
  });
});
