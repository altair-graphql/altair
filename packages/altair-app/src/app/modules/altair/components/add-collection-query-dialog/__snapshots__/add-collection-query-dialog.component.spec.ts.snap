// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCollectionQueryDialogComponent should render correctly 1`] = `
<app-add-collection-query-dialog>
  <app-dialog>
    <div
      class="add-to-collection-wrapper"
    >
      <div
        class="app-dialog-section"
      >
         SAVE_TO_COLLECTION_NAME_OF_QUERY 
        <input
          aria-label="SAVE_TO_COLLECTION_NAME_OF_QUERY"
          class="input ng-untouched ng-pristine ng-valid"
          data-test-id="collection-query-name"
          type="text"
        />
      </div>
      <div
        class="app-dialog-section"
      >
        <nz-form-label
          class="ant-form-item-label ant-col"
        >
          <label>
            SAVE_TO_COLLECTION_SELECT_COLLECTION
            
          </label>
        </nz-form-label>
        <nz-form-control
          class="ant-form-item-control ant-col"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <nz-select
                class="ant-select dialog-block ant-select-in-form-item ant-select-show-arrow ant-select-single ng-untouched ng-pristine ng-valid"
                name="collection"
                nzplaceholder="Choose"
              >
                <nz-select-top-control
                  cdkoverlayorigin=""
                  class="ant-select-selector nz-animate-disabled"
                >
                  <nz-select-search
                    class="ant-select-selection-search"
                  >
                    <input
                      autocomplete="off"
                      class="ant-select-selection-search-input ng-untouched ng-pristine ng-valid"
                      readonly="readonly"
                      style="opacity: 0;"
                    />
                    
                  </nz-select-search>
                  
                  
                  
                  
                  
                  <nz-select-placeholder
                    class="ant-select-selection-placeholder"
                  >
                     Choose 
                    
                    
                  </nz-select-placeholder>
                  
                </nz-select-top-control>
                <nz-select-arrow
                  class="ant-select-arrow"
                >
                  <span
                    class="anticon anticon-down"
                    nz-icon=""
                    nztype="down"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                  
                  
                  
                  
                  
                  
                  
                  
                  
                </nz-select-arrow>
                
                
                
              </nz-select>
            </div>
          </div>
          
          
        </nz-form-control>
      </div>
      
    </div>
  </app-dialog>
</app-add-collection-query-dialog>
`;

exports[`AddCollectionQueryDialogComponent should render correctly with [collections] 1`] = `
<app-add-collection-query-dialog>
  <app-dialog>
    <div
      class="add-to-collection-wrapper"
    >
      <div
        class="app-dialog-section"
      >
         SAVE_TO_COLLECTION_NAME_OF_QUERY 
        <input
          aria-label="SAVE_TO_COLLECTION_NAME_OF_QUERY"
          class="input ng-untouched ng-pristine ng-valid"
          data-test-id="collection-query-name"
          type="text"
        />
      </div>
      <div
        class="app-dialog-section"
      >
        <nz-form-label
          class="ant-form-item-label ant-col"
        >
          <label>
            SAVE_TO_COLLECTION_SELECT_COLLECTION
            
          </label>
        </nz-form-label>
        <nz-form-control
          class="ant-form-item-control ant-col"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <nz-select
                class="ant-select dialog-block ant-select-in-form-item ant-select-show-arrow ant-select-single ng-untouched ng-pristine ng-valid"
                name="collection"
                nzplaceholder="Choose"
              >
                <nz-select-top-control
                  cdkoverlayorigin=""
                  class="ant-select-selector nz-animate-disabled"
                >
                  <nz-select-search
                    class="ant-select-selection-search"
                  >
                    <input
                      autocomplete="off"
                      class="ant-select-selection-search-input ng-untouched ng-pristine ng-valid"
                      readonly="readonly"
                      style="opacity: 0;"
                    />
                    
                  </nz-select-search>
                  
                  
                  
                  
                  
                  <nz-select-placeholder
                    class="ant-select-selection-placeholder"
                  >
                     Choose 
                    
                    
                  </nz-select-placeholder>
                  
                </nz-select-top-control>
                <nz-select-arrow
                  class="ant-select-arrow"
                >
                  <span
                    class="anticon anticon-down"
                    nz-icon=""
                    nztype="down"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                  
                  
                  
                  
                  
                  
                  
                  
                  
                </nz-select-arrow>
                
                
                
              </nz-select>
            </div>
          </div>
          
          
        </nz-form-control>
      </div>
      
    </div>
  </app-dialog>
</app-add-collection-query-dialog>
`;

exports[`AddCollectionQueryDialogComponent should render correctly with [windowTitle] 1`] = `
<app-add-collection-query-dialog>
  <app-dialog>
    <div
      class="add-to-collection-wrapper"
    >
      <div
        class="app-dialog-section"
      >
         SAVE_TO_COLLECTION_NAME_OF_QUERY 
        <input
          aria-label="SAVE_TO_COLLECTION_NAME_OF_QUERY"
          class="input ng-untouched ng-pristine ng-valid"
          data-test-id="collection-query-name"
          type="text"
        />
      </div>
      <div
        class="app-dialog-section"
      >
        <nz-form-label
          class="ant-form-item-label ant-col"
        >
          <label>
            SAVE_TO_COLLECTION_SELECT_COLLECTION
            
          </label>
        </nz-form-label>
        <nz-form-control
          class="ant-form-item-control ant-col"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <nz-select
                class="ant-select dialog-block ant-select-in-form-item ant-select-show-arrow ant-select-single ng-untouched ng-pristine ng-valid"
                name="collection"
                nzplaceholder="Choose"
              >
                <nz-select-top-control
                  cdkoverlayorigin=""
                  class="ant-select-selector nz-animate-disabled"
                >
                  <nz-select-search
                    class="ant-select-selection-search"
                  >
                    <input
                      autocomplete="off"
                      class="ant-select-selection-search-input ng-untouched ng-pristine ng-valid"
                      readonly="readonly"
                      style="opacity: 0;"
                    />
                    
                  </nz-select-search>
                  
                  
                  
                  
                  
                  <nz-select-placeholder
                    class="ant-select-selection-placeholder"
                  >
                     Choose 
                    
                    
                  </nz-select-placeholder>
                  
                </nz-select-top-control>
                <nz-select-arrow
                  class="ant-select-arrow"
                >
                  <span
                    class="anticon anticon-down"
                    nz-icon=""
                    nztype="down"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                  
                  
                  
                  
                  
                  
                  
                  
                  
                </nz-select-arrow>
                
                
                
              </nz-select>
            </div>
          </div>
          
          
        </nz-form-control>
      </div>
      
    </div>
  </app-dialog>
</app-add-collection-query-dialog>
`;
